### PR TITLE
Migrate to GitHub Actions.

### PR DESCRIPTION
**Pull request by jezdez**
Travis CI has a new pricing model which places limits on open source.

* https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
* https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

Many projects are moving to GitHub Actions instead, including Jazzband projects: 

* https://github.com/orgs/jazzband/projects/1
* https://github.com/jazzband-roadies/help/issues/206

This is based on https://github.com/jazzband/contextlib2/pull/26.

It drops support for Python 3.5 because it's EOL.

Python 3.9 support is also added.

TODO:

* [x] @jezdez to add `JAZZBAND_RELEASE_KEY` to the repo secrets.

### CODE REVIEW

**Review by jdufresne — APPROVED** [review]
Thanks!

The dropping of Python 3.5 support should be added to the `CHANGELOG.rst`

**Review by hugovk — APPROVED** [review]
🎈

File: tox.ini [hunk 1]
```diff
@@ -4,27 +4,29 @@ envlist =
...
+    3.8: py38, black, flake8, isort
+    3.9: py39
 
 [testenv]
 deps =
-    django111: Django~=1.11.0
-    django22: Django>=2.2<3.0
-    django30: Django>=3.0<3.1
-    djangomaster: https://github.com/django/django/archive/master.tar.gz#egg=django
+    dj111: Django~=1.11.0
```
Comment (aleksihakli) [inline tox.ini:20]: Django 1.11 is EOL and I think should be dropped from the support matrix in any new releases?

```suggestion
```

File: tox.ini [hunk 2]
```diff
@@ -4,27 +4,29 @@ envlist =
     black
     flake8
     isort
-    py35-django111
-    py{36,37,38}-django{111, 22, 30, master}
+    py{36,37,38,39}-dj{22, 30, 31, master}
```
Comment (aleksihakli) [inline tox.ini:7]: Obsolete spacing in test version definition

```suggestion
    py{36,37,38,39}-dj{22,30,31, master}
```

File: tox.ini [hunk 3]
```diff
@@ -4,27 +4,29 @@ envlist =
...
+    dj30: Django>=3.0,<3.1
+    dj31: Django>=3.1,<3.2
+    djmaster: https://github.com/django/django/archive/master.tar.gz#egg=django
     coverage
 commands =
     coverage run -m django test --settings=tests.settings {posargs}
     coverage report
+    coverage xml
 
 [testenv:black]
```
Comment (aleksihakli) [inline tox.ini:31]: There needs to be an ignore rule for the django master branch tests here as well.

```suggestion
[testenv:py{36,37,38,39,py3}-dj{master}]
ignore_errors = true
ignore_outcome = true

[testenv:black]
```

**Review by aleksihakli — APPROVED** [review]
A few suggestions on the tox.ini configuration.

File: tox.ini (hunk 1 above)
Comment (jezdez) [reply tox.ini:20]: Right, that's actually a leftover anyways, wasn't used in the test matrix anymore.

**Comment by jezdez** [comment]
Thanks all for the reviews! 😍

**Pull request merged**